pf/internal/pfutils/attr.go

```diff
@@ -142,7 +152,7 @@ func LookupTerraformPath(schema Schema, path *tftypes.AttributePath) (LookupResu
 		}
 	}
 
-	return res, fmt.Errorf("LookupTerraformPath failed at path %s, got unexpected %v", path, res)
+	return LookupResult{IsMisc: true}, nil
```

Comment:
The "misc" case such as just a path `"set_optional"."foo"` was resolving to a scalar String but this was coded on the `err` path, which is certainly bad style, lifting this out to a non-err path here with a special misc marker.